.github/workflows/publish-release.yaml

```diff
@@ -0,0 +1,47 @@
+name: Gingersnap db-syncer - Create tag and Publish
+
+on:
+  workflow_call:
+    inputs:
+      ref:
+        type: string
+        default: main
+      repo: 
+        type: string
+        default: gingersnap-project/db-syncer
+      tag:
+        type: string
+        default: latest
+        
+      
+env:
+  GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}  
+
+jobs:
+  tag:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Configure Git
+        run: |
+          git config --global user.name "infinispan-qe-bot"
+          git config --global user.email "q*@infinispan.org"
+
+      - name: Checkout
+        uses: actions/checkout@v3
+        with: 
+          ref: ${{ inputs.ref }} 
+          repository: ${{ inputs.repo }}
+          
+      - name: Tag the release
+        run: |
+         git tag -a v${{ inputs.tag }} -m "Release db-syncer:: ${{ inputs.tag }}"
+         git push origin --tags
+
+  publish:
+    needs: tag
+    uses: ./.github/workflows/publish-image.yaml
+    secrets: inherit
+    with:
+      tag: v${{ inputs.tag }}
```

Comment:
I think we need to add the following inputs to the `publish-image.yaml` so that we can pass  them here and then pass these values to `build-image.yaml` from with `publish-image.yaml`:

```yaml
      ref:
        type: string
        default: ${{ github.ref }}
      repository:
        type: string
        default: ${{ github.repository }}
```

Without this, the default value for `build-image.yaml`'s `ref` and `repository` field will be `{{ github.ref }}` and `{{ github.repository}}` which will be set to the values of the original calling workflow in `gingersnap-project/release`.